experiments/dutch_auctions/dutch_auctions_stats.py

```diff
@@ -46,8 +46,65 @@ def track_stat(state, action, results):
 
 
 def auction_debt():
-    def track_stat(state, action, _results):
+    def track_stat(state, action):
         if action["key"] == "T_START":
             state["stats"]["auction_debt"] = state["dog"].Dirt
 
     return track_stat
+#net difference
+def net_diff():
+    def track_stat(state, results, action):
+        if action["key"] == "T_START":
+            state["stats"]["net_diff"] = Rad(0)
+        elif action["key"] == "BARK":
+            ilk_id = action["args"][0]
+            clip = state["clippers"][ilk_id]
+            tab = results[0]
+            incentive = clip.tip + tab * Rad(clip.chip)
+            state["stats"]["net_diff"] -= incentive
+        elif action["key"] == "REDO":
+            sale_id = action["args"][0]
+            ilk_id = action["extra"]["ilk_id"]
+            clip = state["clippers"][ilk_id]
+            sale = clip.sales[sale_id]
+            incentive = clip.tip + sale.tab * Rad(clip.chip)
+            state["stats"]["net_diff"] -= incentive
+        elif action["key"] == "TAKE":
+            state["stats"]["net_diff"] += results[0]
+    return track_stat
+
+
+#differenceratio
+def net_diff():
+    def track_stat(state, results, action):
```

Comment:
have this stat return the ratio of cumulative incentives paid to cumulative bids placed.
sum into `state["stats"]["difference_ratio"]["cumulative_incentive"]` and `state["stats"]["difference_ratio"]["cumulative_bid"]` every timestep, and save the ratio of the two in `state["stats"]["difference_ratio"]["ratio"]`

Reply:
done